### PR TITLE
fix controller-manager port

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/23435

Fixing incorrect controller-manager port

* keep default behaviour
* retrieve port from annotation if available

### CODE REVIEW

File: files/templates/scrapeconfigs/additional-scrape-configs.template.yaml [hunk 1]
```diff
@@ -255,22 +255,28 @@
   relabel_configs:
   - source_labels: [__address__]
 [[- if .CAPICluster ]]
-    replacement: $1:10257
+    replacement: 10257
```
Comment (QuentinBisson) [inline files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:258]: LGTM but can you keep the instance label anyway ?
Also, should we not do the same for the scheduler ?

**Comment by TheoBrigitte** [comment]
port|release
-----|--------
10251 | 16.5.1
10257 | 18.0.2
10251 | 17.2.0

**Comment by QuentinBisson** [comment]
@TheoBrigitte what's the status on this PR ?

**Comment by TheoBrigitte** [comment]
> @TheoBrigitte what's the status on this PR ?

- tested port fix for controller-manager :heavy_check_mark: 
- implemented fix for scheduler (to be tested)

File: .nancy-ignore [hunk 2]
```diff
@@ -33,3 +33,6 @@ sonatype-2021-1485 until=2022-11-01
 # - sigs.k8s.io/cluster-api@v1.0.5
 # - sigs.k8s.io/controller-runtime@v0.10.3
 CVE-2021-20329 until=2022-11-01
+
+# pkg:golang/github.com/labstack/echo/v4@v4.5.0
+sonatype-2022-5436
```
Comment (QuentinBisson) [inline .nancy-ignore:38]: I think you can fix it by upgrading the apimachinery depencency to 0.24.0

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- Fix controller manager port to be dynamic.
```
Comment (QuentinBisson) [inline CHANGELOG.md:12]: Is it dynamic or taken from an annotation ?

File: files/templates/scrapeconfigs/additional-scrape-configs.template.yaml [hunk 4]
```diff
@@ -255,22 +255,32 @@
...
 [[- else ]]
-    replacement: $1:10252
+    replacement: 10252
 [[- end ]]
+    target_label: __port
```
Comment (QuentinBisson) [inline files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:262]: Would it make sense to rename it to __tmp_port ?

Also do not forget to use a label drop to remove the label at the end

Comment (marieroque) [reply files/templates/scrapeconfigs/additional-scrape-configs.template.yaml:262]: Ok we rename by __tmp_port.

BTW, label drop is not necessary :
"Labels starting with __ will be removed from the label set after target relabeling is completed." [prometheus doc](https://prometheus.io/docs/prometheus/latest/configuration/configuration/)

File: .nancy-ignore (hunk 2 above)
Comment (marieroque) [reply .nancy-ignore:38]: We'll do the fix in another PR :-)

**Review by QuentinBisson — APPROVED** [review]
🚀

**Pull request merged**